File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        s = ""
        n = len(strs)

        if n == 0:
            return ""
        
        for i in range(0, len(strs)):
            s += strs[i] + "-"
         
        return s

    def decode(self, s: str) -> List[str]:
        res = []

        n = len(s)
        temp = ""

        if n == 0:
            return []

        for i in range(0, n-1):
            if s[i] != '-':
                temp += s[i]
            
            else:
                res.append(temp)
                temp = ""
            
        res.append(temp)
        return res

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        map<vector<int>, vector<string>> mp;

        for(int i = 0 ; i < n ; i++){
            vector<int> freq(26, 0);
            for(int j = 0 ; j < strs[i].length(); j++){
                freq[strs[i][j] - 'a']++;
            }

            mp[freq].push_back(strs[i]);
        }

        vector<vector<string>> res;

        for(auto it:mp) {
            res.push_back(it.second);
        }

        return res;
    }
};
